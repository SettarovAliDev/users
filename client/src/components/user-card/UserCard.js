import React from "react";
import { useNavigate } from "react-router-dom";

import {
  UserCardContainer,
  UserCardHeading,
  UserCardSpan,
} from "./UserCardStyles";

const UserCard = ({ user }) => {
  const navigate = useNavigate();
  return (
    <UserCardContainer onClick={() => navigate(`${user.id}`)}>
      <UserCardHeading>{user.username}</UserCardHeading>
      <UserCardSpan>{user.email}</UserCardSpan>
      <UserCardSpan>
        {Object.values(user.profiles).length} profiles
      </UserCardSpan>
    </UserCardContainer>
  );
};

export default UserCard;
