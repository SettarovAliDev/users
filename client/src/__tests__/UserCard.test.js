import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { render, fireEvent, screen } from '@testing-library/react';
import 'jest-styled-components';
import '@testing-library/jest-dom/extend-expect';

import UserCard from '../components/user-card/UserCard';
import { usersData } from './test-utils';

const mockedUsedNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedUsedNavigate,
}));

const renderWithRouter = (component) => {
  return {
    ...render(<MemoryRouter>{component}</MemoryRouter>),
  };
};

const user = usersData[0];

it('should render UserCard correctly', () => {
  const tree = renderer.create(<UserCard user={user} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('should equal to user fields', () => {
  const { getByTestId } = render(<UserCard user={user} />);
  expect(getByTestId('username')).toHaveTextContent(user.username);
  expect(getByTestId('email')).toHaveTextContent(user.email);
  expect(getByTestId('profiles')).toHaveTextContent(
    `${Object.values(user.profiles).length} profiles`
  );
});

it('should navigate to user page', () => {
  const { getByTestId } = renderWithRouter(<UserCard user={user} />);
  fireEvent.click(getByTestId('card'));
  expect(mockedUsedNavigate).toHaveBeenCalledWith(`${user.id}`);
});
