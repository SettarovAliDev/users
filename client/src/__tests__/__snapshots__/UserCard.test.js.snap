// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render UserCard correctly 1`] = `
.c0 {
  width: calc((100% - 3.2rem * 3) / 4);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 3rem 5rem;
  border-radius: 1.6rem;
  background-color: var(--background-color-white);
  color: var(--text-color-dark-2);
  cursor: pointer;
}

.c0:hover {
  background-color: #dfffdf;
}

.c0 > * {
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
  width: 100%;
  text-align: center;
}

.c0 > *:not(:last-child) {
  margin-bottom: 2rem;
}

.c1 {
  font-size: 2.4rem;
  font-weight: 500;
  -webkit-letter-spacing: 0.75px;
  -moz-letter-spacing: 0.75px;
  -ms-letter-spacing: 0.75px;
  letter-spacing: 0.75px;
}

.c2 {
  font-size: 2.4rem;
  font-weight: 400;
  -webkit-letter-spacing: 0.75px;
  -moz-letter-spacing: 0.75px;
  -ms-letter-spacing: 0.75px;
  letter-spacing: 0.75px;
}

@media screen and (max-width:1679px) {
  .c0 {
    width: calc((100% - 3.2rem * 2) / 3);
    min-width: 39.6rem;
  }
}

@media screen and (max-width:1272px) {
  .c0 {
    width: calc((100% - 3.2rem) / 2);
    min-width: 39.6rem;
  }
}

@media screen and (max-width:860px) {
  .c0 {
    width: 100%;
    min-width: 39.6rem;
  }
}

<li
  className="c0"
  data-testid="card"
  onClick={[Function]}
>
  <h3
    className="c1"
    data-testid="username"
  >
    ali
  </h3>
  <span
    className="c2"
    data-testid="email"
  >
    ali@gmail.com
  </span>
  <span
    className="c2"
    data-testid="profiles"
  >
    2
     profiles
  </span>
</li>
`;
